src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3839,93 +3840,111 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       val annTypeSym = annType.typeSymbol
       val isJava = annType != null && annTypeSym.isJavaDefined
 
-      finish(
-        if (typedFun.isErroneous || annType == null)
-          ErroneousAnnotation
-        else if (isJava || annTypeSym.isNonBottomSubClass(ConstantAnnotationClass)) {
-          // Arguments of Java annotations and ConstantAnnotations are checked to be constants and
-          // stored in the `assocs` field of the resulting AnnotationInfo
-          if (argss.lengthIs > 1) {
-            reportAnnotationError(MultipleArgumentListForAnnotationError(ann))
-          } else {
-            // TODO: annType may have undetermined type params -- can we infer them?
-            val annScopeJava =
-              if (isJava) annType.decls.filter(sym => sym.isMethod && !sym.isConstructor && sym.isJavaDefined)
-              else EmptyScope // annScopeJava is only used if isJava
-
-            val names = mutable.Set[Symbol]()
-            names ++= (if (isJava) annScopeJava.iterator
-                       else typedFun.tpe.params.iterator)
-
-            def hasValue = names exists (_.name == nme.value)
-            val namedArgs = argss match {
-              case List(List(arg)) if !isNamedArg(arg) && hasValue => gen.mkNamedArg(nme.value, arg) :: Nil
-              case List(args) => args
-            }
+      def constantly = {
+        // Arguments of Java annotations and ConstantAnnotations are checked to be constants and
+        // stored in the `assocs` field of the resulting AnnotationInfo
+        if (argss.lengthIs > 1) {
+          reportAnnotationError(MultipleArgumentListForAnnotationError(ann))
+        } else {
+          // TODO: annType may have undetermined type params -- can we infer them?
+          val annScopeJava =
+            if (isJava) annType.decls.filter(sym => sym.isMethod && !sym.isConstructor && sym.isJavaDefined)
+            else EmptyScope // annScopeJava is only used if isJava
+
+          val names = mutable.Set[Symbol]()
```

Comment:
The class `Symbol` includes an instance-unique `id` value, of type `Int`. Instead of using a Set of symbols, could we use a set of their `id`s? Sets of integers have efficient implementations.

Reply:
A sage conjecture.